spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java

```diff
@@ -128,6 +124,15 @@ public void setApplicationContext(ApplicationContext context) throws BeansExcept
 		return builder;
 	}
 
+	private Logger getLogger(FeignContext context) {
+		Logger logger = getOptional(context, Logger.class);
```

Comment:
I think if we are going to provide a factory to create the logger, let the factory create the logger.  How that logger is created can be an implementation detail.  (IE if a logger bean is present use it, if not than create a new one)